### ISSUE TITLE
Image Automation Controller is reformatting inline blocks in a single line

### ISSUE THREAD

**Issue by ruiengana** [fluxcd/image-automation-controller#528]
When a manifest has block configuration like below and is updated by image automation controller, blocks are overwritten as a single line of code making it unreadable. Code blocks should not be reformatted.

For example, this is pipe block
```
  config: |
    _format_version: "2.1"
    _transform: true
    services:
    - name: example
      url: https://example.org
      routes:
      - name: example
        paths:
        - /example
```

After image automation controller commit becomes

```
  config: "_format_version: \"2.1\"\n    _transform: true\n    services:\n    - name: example\n      url: https://example.org\n      routes:\n      - name: example\n        paths:\n        - /example\n"
```

This behaviour is unexpected, we would like blocks to stay exactly as they were before.

**Comment by stefanprodan** [comment]
> This behaviour is unexpected, we would like blocks to stay exactly as they were before.

We use kustomize kyaml to mutate the YAMLs and this is how it behaves. I suggest you move the `images` block that's being patched, to a Flux Kustomization e.g.

```yaml
apiVersion: kustomize.toolkit.fluxcd.io/v1
kind: Kustomization
metadata:
  name: podinfo
  namespace: default
spec:
  images:
    - name: ghcr.io/stefanprodan/podinfo
      newName: ghcr.io/stefanprodan/podinfo # {"$imagepolicy": "flux-system:podinfo:name"}
      newTag: 5.0.0 # {"$imagepolicy": "flux-system:podinfo:tag"}
```

**Comment by ruiengana** [comment]
> We use kustomize kyaml to mutate the YAMLs and this is how it behaves. I suggest you move the `images` block that's being patched, to a Flux Kustomization e.g.
> 
We use the image automation controller mainly with HelmReleases managed by Helm Controller to override the images and tags values from helm.

I don't think I understood how I could apply your solution to it.

**Comment by stefanprodan** [comment]
You can create a kustomizeconfig to teach Kustomize how to use `images` in custom resources, docs here https://github.com/kubernetes-sigs/kustomize/blob/master/examples/transformerconfigs/images/README.md

**Comment by ruiengana** [comment]
Also, we have annotations with code blocks too in the same YAML and curiously the annotations part is left exactly as it was before.

**Comment by ruiengana** [comment]
> We use kustomize kyaml to mutate the YAMLs and this is how it behaves. I suggest you move the `images` block that's being patched, to a Flux Kustomization e.g.
> 
We also have annotations with code blocks in the same YAML and interestingly the annotations are being left intact. Can you help me understand why code blocks in annotations are treated differently?

**Comment by Orpere** [comment]
@stefanprodan The config @ruiengana refers to is on a HelmRelease and is part of the helm values;
I wonder if you can clarify your answers, please  
Do you know if there is something I can't figure out here?

**Comment by stefanprodan** [comment]
Another solution is to move the `config` to a ConfigMap and use `valuesFrom` in the HelmRelease, so when Flux patches the images, it will not mess up the config as it's not there.

**Comment by ruiengana** [comment]
> Another solution is to move the `config` to a ConfigMap and use `valuesFrom` in the HelmRelease, so when Flux patches the images, it will not mess up the config as it's not there.

Yes, I think that will work.

Still unsure why inline blocks in annotations are not being messed up while other inline blocks are. Isn't that strange?

**Comment by ruiengana** [comment]
> > Still unsure why inline blocks in annotations are not being messed up while other inline blocks elsewhere are. Isn't that strange?
> 
> It is, I can't find the issue now, but it's here somewhere https://github.com/kubernetes-sigs/kustomize/issues?q=is%3Aissue+is%3Aopen+ as I said, we don't edit the YAMLs with our code, we use Kustomize kyaml in Flux.

I can provide a HelmRelease with a simple deployment to demonstrate. Let me know if that helps?

**Comment by ruiengana** [comment]
@stefanprodan this is becoming a real problem for us. We have large inline block pieces being reformatted in a single line and messing everything :( we currently reverting the commit side effect manually which isn't sustainable. Do you know if this is already open in [kustomize ](https://github.com/kubernetes-sigs/kustomize/issues?q=is%3Aissue+is%3Aopen+) ?

**Comment by ruiengana** [comment]
@stefanprodan any update on this?

**Comment by AchazRyus** [comment]
This issue is still relevant for me also. Would love to hear any update !

**Comment by stefanprodan** [comment]
This should've been fixed in https://github.com/fluxcd/image-automation-controller/pull/605

You can try it out using `ghcr.io/fluxcd/image-automation-controller:preview-13413b26`

**Comment by AchazRyus** [comment]
I just tried out and unfortunately it's not resolving the problem.
I used `ghcr.io/fluxcd/image-automation-controller:preview-13413b26` and `https://github.com/fluxcd-community/helm-charts/releases/tag/flux2-2.12.0` for the version of the other components.

**Comment by AchazRyus** [comment]
Great ! Thanks you

**Comment by stefanprodan** [comment]
This should be fixed in Flux 2.2.0, please open a new issue if that's not the case.
